src/compiler/checker.ts

```diff
@@ -18046,8 +18260,16 @@ namespace ts {
             return !!(getObjectFlags(type) & ObjectFlags.Reference && (<TypeReference>type).target.objectFlags & ObjectFlags.Tuple);
         }
 
+        function isGenericTupleType(type: Type): type is TupleTypeReference {
+            return isTupleType(type) && !!(type.target.combinedFlags & ElementFlags.Variadic);
+        }
+
+        function isSingleElementGenericTupleType(type: Type): type is TupleTypeReference {

```

Comment:
Same re: type guardness

Reply:
Hmm, not crazy about removing these or introducing types like `SingleElementGenericTupleTypeReference` that we don't otherwise need.